File: backend/src/main/java/net/tape/model/Host.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class Host implements Serializable {
    private Long id;
    private Long showId;
    private String initials;
    private String name;
    private String role;
    private int sort;
}

File: web/src/api/components/schemas/Host.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type Host = {
	id: Option<Int>;
	showId: Option<Int>;
	initials: Option<string>;
	name: Option<string>;
	role: Option<string>;
	sort: Option<Int>;
};
export const HostIO = type(
	{
		id: optionFromNullable(Int),
		showId: optionFromNullable(Int),
		initials: optionFromNullable(string),
		name: optionFromNullable(string),
		role: optionFromNullable(string),
		sort: optionFromNullable(Int),
	},
	'Host',
);
